apps/src/JSInterpreter.js

```diff
@@ -324,6 +326,17 @@ JSInterpreter.prototype.removeStoppedAtBreakpointRowForScope = function (scope)
 };
 
 /**
+ * Determines if the program is done executing.
+ *
+ * @return {boolean} true if program is complete (or an error has occurred).
+ */
+JSInterpreter.prototype.isProgramDone = function () {
+  return this.executionError ||
+      !this.interpreter ||
+      !this.interpreter.stateStack.length;
+};
```

Comment:
If `this.executionError` is taken out of instance state, could we signal "done" by `executeInterpreter()` returning false?